CRM/Price/BAO/LineItem.php

```diff
@@ -914,19 +914,21 @@ protected function getLineItemsToAlter($submittedLineItems, $entityID, $entity)
       // check through the submitted items if the previousItem exists,
       // if found in submitted items, do not use it for new item creations
       if (in_array($previousLineItem['price_field_value_id'], $submittedPriceFieldValueIDs)) {
+        $submittedLineItem = $submittedLineItems[$previousLineItem['price_field_value_id']];
         // if submitted line items are existing don't fire INSERT query
         if ($previousLineItem['line_total'] != 0) {
           unset($lineItemsToAdd[$previousLineItem['price_field_value_id']]);
         }
         else {
-          $lineItemsToAdd[$previousLineItem['price_field_value_id']]['skip'] = TRUE;
+          $submittedLineItem['skip'] = TRUE;
         }
         // for updating the line items i.e. use-case - once deselect-option selecting again
-        if (($previousLineItem['line_total'] != $submittedLineItems[$previousLineItem['price_field_value_id']]['line_total']) ||
-          ($submittedLineItems[$previousLineItem['price_field_value_id']]['line_total'] == 0 && $submittedLineItems[$previousLineItem['price_field_value_id']]['qty'] == 1) ||
-          ($previousLineItem['qty'] != $submittedLineItems[$previousLineItem['price_field_value_id']]['qty'])
+        if (($previousLineItem['line_total'] != $submittedLineItem['line_total'])
+          ||
```

Comment:
should this go on the previous line?

Reply:
not sure - at some point I liked it better on the next line - it was a deliberate change - but may I was just in a 'next liney mood'

Reply:
Minor aesthetic, but seems kind of weird to me to have the `||` on a line by itself :

![capture d ecran de 2017-11-15 10-46-34](https://user-images.githubusercontent.com/254741/32845254-4b7bded2-c9f2-11e7-8f36-0fbd16ef7dd0.png)

Reply:
I don't know if it 'looks' right - but the change I just pushed is the most readable to me - ie. having the inners of the OR's separated out